rust/candid_derive/src/lib.rs

```diff
@@ -32,11 +34,34 @@ pub(crate) fn idl_hash(id: &str) -> u32 {
 }
 
 #[cfg(feature = "cdk")]
-pub(crate) fn candid_path() -> proc_macro2::TokenStream {
-    quote::quote! { ::ic_cdk::export::candid }
+pub(crate) fn candid_path(
+    custom_candid_path: &Option<proc_macro2::TokenStream>,
+) -> proc_macro2::TokenStream {
+    match custom_candid_path {
+        Some(custom_candid_path_value) => custom_candid_path_value.clone(),
+        None => quote::quote! { ::ic_cdk::export::candid },
+    }
 }
 
 #[cfg(not(feature = "cdk"))]
-pub(crate) fn candid_path() -> proc_macro2::TokenStream {
-    quote::quote! { ::candid }
+pub(crate) fn candid_path(
+    custom_candid_path: &Option<proc_macro2::TokenStream>,
+) -> proc_macro2::TokenStream {
+    match custom_candid_path {
+        Some(custom_candid_path_value) => custom_candid_path_value.clone(),
+        None => quote::quote! { ::candid },
+    }
+}
+
+// TODO I am not sure I parsed this in the best way possible...any pointers would be great
+fn get_custom_candid_path(input: &syn::DeriveInput) -> Option<proc_macro2::TokenStream> {
```

Comment:
It's better to return `Result<TokenStream>` so that we can customize the error message, see https://github.com/dfinity/candid/blob/master/rust/candid_derive/src/func.rs#L215

At the call site, we can convert `syn::Error` to a `compile_error`, see https://github.com/dfinity/candid/blob/master/rust/candid_derive/src/lib.rs#L17

Reply:
Awesome, I have addressed this